hist/hist/test/CMakeLists.txt

```diff
@@ -26,3 +26,7 @@ if(clad)
   ROOT_ADD_GTEST(TFormulaGradientTests TFormulaGradientTests.cxx LIBRARIES Core MathCore Hist)
   ROOT_ADD_GTEST(TFormulaHessianTests TFormulaHessianTests.cxx LIBRARIES Core MathCore Hist)
 endif()
+
+# Regression test for https://github.com/root-project/root/issues/13691
```

Comment:
Maybe a naive question, but why the test in hist/hist and not in core?

Reply:
Totally not a naive question, and the reason is imho extremely convoluted. It is explained both in the commit message as well as in the test file itself. You can find the full description of the debugging steps that took me to the current situation in this PR, starting from [here](https://github.com/root-project/root/pull/13692#issuecomment-1730092957). I can also add the description of the situation to the `CMakeLists.txt` too. In general, we need the test to link to some library outside of `core`, so I chose `hist` absolutely arbitrarily because it's always present in every build. The choice can be changed of course if there is a better place to write the test which still makes it fail as expected

Reply:
I also added some wording to explain better the test placement in the CMakeLists.txt file

Reply:
Note that it might make sense to move the test to roottest.